components.go

```diff
@@ -0,0 +1,97 @@
+package discordgo
+
+import (
+	"encoding/json"
+)
+
+// ComponentType is type of component.
+type ComponentType uint
+
+// Component types.
+const (
+	ActionsRowComponent ComponentType = iota + 1
+	ButtonComponent
+)
+
+// Component is a base interface for all components
+type Component interface {
+	json.Marshaler
+	Type() ComponentType
+}
+
+// ActionsRow is a container for components within one row.
+type ActionsRow struct {
+	Components []Component `json:"components"`
+}
```

Comment:
I was reading the docs and it was clearly saying "Actions row", but maybe we should name it ActionsBar, or stick to the docs? :thinking: 
```suggestion
// ActionsRow is a container for components within one row.
type ActionsRow struct {
	Components []Component `json:"components"`
}
```